tests/functional/object/mcg/test_noobaa_use_new_internal_cert_after_rotation.py

```diff
@@ -0,0 +1,116 @@
+from datetime import datetime
+import logging
+
+from ocs_ci.ocs.constants import SECRET, NOOBAA_S3_SERVING_CERT
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.framework.pytest_customization.marks import (
+    tier1,
+    polarion_id,
+    bugzilla,
+    skipif_ocs_version,
+    red_squad,
+    mcg,
+    config,
+)
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.pod import get_noobaa_endpoint_pods
+from ocs_ci.utility.retry import retry
+
+
+logger = logging.getLogger(__name__)
+
+
+@retry(UnexpectedBehaviour, tries=10, delay=3, backoff=1)
+def get_validity_time_of_certificate(noobaa_endpoint_pods, cmd, old_validity=None):
+    """
+    Number of attempts to retry to get the validity of certificate
+
+    """
+
+    new_cmd_output = noobaa_endpoint_pods[0].exec_sh_cmd_on_pod(command=cmd)
+    # Get the validity time of certificate
+    cmd_output = new_cmd_output.split(",")
+    cmd_output = cmd_output[0].split("\n")
+    new_validity = cmd_output[7].split(",")[0] + cmd_output[8].split(",")[0]
+    if new_validity == old_validity:
+        logger.warn(
+            f"New certificate not created post the deletion of secret {NOOBAA_S3_SERVING_CERT}, retrying again"
+        )
+        raise UnexpectedBehaviour(
+            f"New certificate not created post the deletion of secret {NOOBAA_S3_SERVING_CERT}"
+        )
+    logger.info("New certificate created")
+    return new_validity, new_cmd_output
+
+
+@mcg
+@red_squad
+@tier1
+@skipif_ocs_version("<4.12")
+@bugzilla("2237903")
+@polarion_id("OCS-6191")
+class TestNoobaaUseNewInternalCertAfterRotation:
+    def test_noobaa_use_new_internal_cert_after_rotation(self):
+        """
+
+        Objective of this test is to verify that noobaa use the new internal certificate after rotation
+
+        """
+
+        # Run command to get the validity of certificate before deleting secret
```

Comment:
if you implement the changes suggested by me above, you can use the func `def get_validity_time_of_certificate(noobaa_endpoint_pods):` here